src/functions_math.jl

```diff
@@ -108,3 +113,131 @@ function symmetric_names(L::Tuple{Symbol,Symbol}, dims::Integer)
     end
     return compile_time_return_hack(names)
 end
+
+################################################
+# FFT
+
+for fun in (:fft, :ifft, :bfft)
+    plan_fun = Symbol(:plan_, fun)
+    @eval begin
+
+        function AbstractFFTs.$fun(A::NamedDimsArray{L}) where {L}
+            data = AbstractFFTs.$fun(parent(A))
+            return NamedDimsArray(data, wave_name(L))
+        end
+
+        function AbstractFFTs.$fun(A::NamedDimsArray{L,T,N}, dims) where {L,T,N}
+            numerical_dims = dim(A, dims)
+            data = AbstractFFTs.$fun(parent(A), numerical_dims)
+            newL = wave_name(L, numerical_dims)
+            return NamedDimsArray(data, newL)
+        end
+
+        """
+            $($fun)(A, :time => :freq, :x => :kx)
+
+        Acting on a `NamedDimsArray`, this specifies to take the transform along the dimensions
+        named `:time, :x`, and return an array with names `:freq` and `:kx` in their places.
+
+        If new names are not given, then the default is `:x => :x∿` and `:x∿ => :x`,
+        applied to all dimensions, or to those specified as usual, e.g. `$($fun)(A, (1,2))`
+        or `$($fun)(A, :time)`. The symbol "∿" can be typed by `\\sinewave<tab>`.
+        """
+        function AbstractFFTs.$fun(A::NamedDimsArray{L,T,N}, p::Pair{Symbol,Symbol}, ps::Pair{Symbol,Symbol}...) where {L,T,N}
```

Comment:
Thoughs on the pro and con of using kwargs to rename?
```
fft((A;  x=:xf, y=:yf)
```

Reply:
I thought about that, but concluded that it was too confusing about which one is the old & the new name.